### PR TITLE
zibs MedicalDevice HearingFunction and VisualFunction

### PR DESCRIPTION
**Pull request by ArdonToonstra**
Please review these profiles. I tried to improve the mapping of MedicalDevice ProductID in FHIR, which still has an open question (https://bits.nictiz.nl/browse/ZIB-1535). I haven't slilenced all issues for ProductId in the known-issues for this yet.

Moreover, I believe the QA tooling has some complaints that will need to be silenced in the tooling:

>     == NL-CM:4.16.4 (DeviceUseStatement)
>       short:       WARN ('MedicalDevice / VisualAid' instead of 'MedicalDevice / VisualAid:MedicalDevice')
>      datatype:    ERROR (undefined instead of undefined)

### CODE REVIEW

**Comment by pieter-edelman-nictiz** [comment]
I'd like to propose other names for these profiles using the zib concept names rather than the FHIR resource names. I think this is more clear than some part zib concept, some part FHIR resource:
* zib-MedicalDevice-Device => zib-MedicalDevice.Product
* zib-HearingFunction-MedicalDevice => zib-HearingFunction.HearingAid
* zib-HearingFunction-MedicalDevice => zib-HearingFunction.HearingAid.Product
* zib-VisualFunction-MedicalDevice => zib-VisualFunction.VisualAid
* zib-VisualFunction-MedicalDevice-Device => zib-VisualFunction.VisualAid.Product

File: resources/zib/zib-MedicalDevice.xml [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        <targetProfile value="http://hl7.org/fhir/StructureDefinition/Media" />
+        <targetProfile value="http://nictiz.nl/fhir/StructureDefinition/zib-Problem" />
+      </type>
+      <mapping>
+        <identity value="zib-medicaldevice-v3.3.1-2020EN" />
+        <map value="NL-CM:10.1.7" />
+        <comment value="Indication" />
+      </mapping>
+    </element>
+    <element id="DeviceUseStatement.bodySite">
```
Comment (pieter-edelman-nictiz) [inline resources/zib/zib-MedicalDevice.xml:149]: Binding to http://decor.nictiz.nl/fhir/ValueSet/2.16.840.1.113883.2.4.3.11.60.121.11.11--20200901000000 is required here.

**Comment by ArdonToonstra** [comment]
> I'd like to propose other names for these profiles using the zib concept names rather than the FHIR resource names. I think this is more clear than some part zib concept, some part FHIR resource:
> 
> * zib-MedicalDevice-Device => zib-MedicalDevice.Product
> * zib-HearingFunction-MedicalDevice => zib-HearingFunction.HearingAid
> * zib-HearingFunction-MedicalDevice => zib-HearingFunction.HearingAid.Product
> * zib-VisualFunction-MedicalDevice => zib-VisualFunction.VisualAid
> * zib-VisualFunction-MedicalDevice-Device => zib-VisualFunction.VisualAid.Product
> 
> And than the same pattern for nl-core

I agree! Fixed.

File: resources/zib/zib-MedicalDevice.xml (hunk 1 above)
Comment (ArdonToonstra) [reply resources/zib/zib-MedicalDevice.xml:149]: I am not sure which valueset you mean?
* MedicalDevice does not define a ValueSet for AnatomicalLocation. 
* Can't find this valueset in the zib AnatomicalLocation
* Can find this valueset in HearingFunction or VisualFunction (and Art-Decor is down)

Comment (pieter-edelman-nictiz) [reply resources/zib/zib-MedicalDevice.xml:149]: It's the ValueSet from zib AnatomicalLocatuion. See the profilers guidance in het zib-AnatomicalLocation datatype profile.

**Comment by pieter-edelman-nictiz** [comment]
The errors:
```== NL-CM:4.16.4 (DeviceUseStatement)
  short:       WARN ('MedicalDevice / VisualAid' instead of 'MedicalDevice / VisualAid:MedicalDevice')
 datatype:    ERROR (undefined instead of undefined)
```

are now changed to:

```
     == NL-CM:4.16.4 (DeviceUseStatement)
        short:       WARNING (MedicalDevice / VisualAid instead of MedicalDevice / VisualAid:MedicalDevice)
        datatype:    WARNING (undefined instead of Reference)
```

I think it's correct to conclude that the short should be combined; there's a mapping to both elements. Regarding the inclusion of "VisualAid:" ... that's a zib bug (single colon instead of double colon, so splitting doesn't work correctly).

File: resources/zib/zib-MedicalDevice.xml (hunk 1 above)
Comment (ArdonToonstra) [reply resources/zib/zib-MedicalDevice.xml:152]: Fixed!

**Comment by ArdonToonstra** [comment]
Sorry I missed updating the names, titles etc too! I now also have updated the know-issues.yml to the new names and added more to it. 

The only remaining errors are the below ones.

==== snapshots/zib-HearingFunction.HearingAid.json
     == NL-CM:10.1.1 (DeviceUseStatement)
        datatype:    ERROR (undefined instead of undefined)
     == NL-CM:4.17.4 (DeviceUseStatement)
        datatype:    ERROR (undefined instead of undefined)

File: resources/zib/zib-MedicalDevice.Product.xml [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+      <path value="Device.udiCarrier.carrierHRF" />
+      <short value="ProductID" />
+      <alias value="ProductID" />
+      <mapping>
+        <identity value="zib-medicaldevice-v3.3.1-2020EN" />
+        <map value="NL-CM:10.1.4" />
+        <comment value="ProductID" />
+      </mapping>
+    </element>
+    <element id="Device.manufactureDate">
```
Comment (LuudSlagter) [inline resources/zib/zib-MedicalDevice.Product.xml:111]: Should this element be mapped to NL-CM:10.1.4 as well (compare with expirationDate)?

File: resources/zib/zib-MedicalDevice.Product.xml [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+  <abstract value="true" />
+  <type value="Device" />
+  <baseDefinition value="http://hl7.org/fhir/StructureDefinition/Device" />
+  <derivation value="constraint" />
+  <differential>
+    <element id="Device">
+      <path value="Device" />
+      <short value="Product" />
+      <definition value="The medical device (internally or externally)." />
+      <comment value="The zib requires the coding systems Health Industry Bar Code (HIBC) or the GS1's Global Trade Item Number (GTIN) to indentify devices. The device identification happens on the basis of an UDI. This resources contains a mapping on all UDI related elements. These elements, which are also relavant and populated when GTIN is used, contain a mapping to the zib concept ProductID.&#xD;&#xA;&#xD;&#xA;In short, the full indentifier will be populated in `udiCarrier.carrierHRF` and may be broken down into structural parts. These parts are the `.identifier`, `.udiCarrier.deviceIdentifier`, `.manufactureDate`, `.expirationDate`, `.lotNumber` and `.serialNumber`.&#xD;&#xA;&#xD;&#xA;The element `udiCarrier.issuer` is used to indicate which 'coding system' is used." />
```
Comment (LuudSlagter) [inline resources/zib/zib-MedicalDevice.Product.xml:36]: Replace 'relavant' by 'relevant'.

File: resources/zib/zib-VisualFunction.VisualAid.xml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      <path value="DeviceUseStatement.device" />
+      <type>
+        <code value="Reference" />
+        <targetProfile value="http://hl7.org/fhir/StructureDefinition/Device" />
+        <targetProfile value="http://nictiz.nl/fhir/StructureDefinition/zib-VisualFunction.VisualAid.Product" />
+      </type>
+    </element>
+    <element id="DeviceUseStatement.reasonReference">
+      <path value="DeviceUseStatement.reasonReference" />
+      <definition value="The zib HearingFunction Observation resource whose existence justifies this DeviceUseStatement. This element is also used to refer to the medical reason for use of the medical device." />
```
Comment (LuudSlagter) [inline resources/zib/zib-VisualFunction.VisualAid.xml:53]: 'HearingFunction' should be replaced by 'VisualFunction'.

**Review by pieter-edelman-nictiz — CHANGES_REQUESTED** [review]

File: resources/zib/zib-VisualFunction.xml [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        <comment value="VisualFunction" />
+      </mapping>
+    </element>
+    <element id="Observation.note">
+      <path value="Observation.note" />
+      <short value="Comment" />
+      <definition value="A comment on the visual function." />
+      <alias value="Toelichting" />
+      <max value="1" />
+      <mapping>
```
[the remark this answers is not in the document]
Comment (LuudSlagter) [reply resources/zib/zib-VisualFunction.xml:85]: Fixed!

File: resources/zib/zib-HearingFunction.HearingAid.xml [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+      </type>
+    </element>
+    <element id="DeviceUseStatement.reasonReference">
+      <path value="DeviceUseStatement.reasonReference" />
+      <definition value="The zib HearingFunction Observation resource whose existence justifies this DeviceUseStatement. This element is also used to refer to the medical reason for use of the medical device." />
+    </element>
+    <element id="DeviceUseStatement.reasonReference:observation">
+      <path value="DeviceUseStatement.reasonReference" />
+      <sliceName value="observation" />
+      <requirements value="The zib HearingFunction Observation resource whose existence justifies this DeviceUseStatement." />
```
[the remark this answers is not in the document]
Comment (LuudSlagter) [reply resources/zib/zib-HearingFunction.HearingAid.xml]: This was already present before this element was sliced, so I copied it to this slice. In retrospect I agree that it doesn't add much, so I have removed it (same for the other unnecessary texts).

**Comment by pieter-edelman-nictiz** [comment]
Manually rebased and merged

**Pull request closed without merging**